take-screenshot/src/main/java/com/esri/arcgisruntime/sample/takescreenshot/MainActivity.java

```diff
@@ -0,0 +1,189 @@
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        // inflate MapView from layout
+        mMapView = (MapView) findViewById(R.id.mapView);
+        // create a map with the BasemapType Imagery with Labels
+        ArcGISMap mMap = new ArcGISMap(Basemap.createImageryWithLabels());
+        // set the map to be displayed in this view
+        mMapView.setMap(mMap);
+
+
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.menu_main, menu);
+
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // handle menu item selection
+
+        int itemId = item.getItemId();
+        if (itemId == R.id.CaptureMap) {
+            // take a screenshot of the image
+            captureScreenshotAsync();
+        }
+
+        return true;
+    }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+        mMapView.pause();
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        mMapView.resume();
+    }
+
+    /**
+     * capture the map as an image
+     */
+    private void captureScreenshotAsync() {
+
+        // export the image from the mMapView
+        final ListenableFuture<Bitmap> export = mMapView.exportImageAsync();
+        export.addDoneListener(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    Bitmap currentMapImage = export.get();
+                    // play the camera shutter sound
+                    MediaActionSound sound = new MediaActionSound();
+                    sound.play(MediaActionSound.SHUTTER_CLICK);
+                    Log.d(TAG,"Captured the image!!");
+                    // save the exported bitmap to an image file
+                    SaveImageTask saveImageTask = new SaveImageTask();
+                    saveImageTask.execute(currentMapImage);
+                } catch (Exception e) {
+                    Log.d(TAG, "Fail to export map image: " + e.getMessage());
+                }
+            }
+        });
+    }
+
+    /**
+     * save the bitmap image to file and open it
+     *
+     * @param bitmap
+     * @throws IOException
+     */
+    private File saveToFile(Bitmap bitmap) throws IOException {
+
+        // create a directory ArcGIS to save the file
+        File root = null;
+        String fileName = "map-export-image" + System.currentTimeMillis() + ".png";
+        root = Environment.getExternalStorageDirectory();
+        File fileDir = new File(root.getAbsolutePath() + "/ArcGIS/");
+        fileDir.mkdirs();
+
+        // create the file inside the directory
+        File file = new File(fileDir, fileName);
+
+        // write the bitmap to PNG file
+        FileOutputStream fos = new FileOutputStream(file);
```

Comment:
This is causing `File` to be null and crashing the app with a `NullPointerException` on line 182. Commenting the `FileOutputStream` avoids the crash.  Please investigate the cause and determine if this is a required fix or whether we can return the `File` as is.